src/libsyntax/parse/parser.rs

```diff
@@ -5809,12 +5809,19 @@ impl<'a> Parser<'a> {
             }
             token::CloseDelim(token::Brace) => {}
             token::DocComment(_) => {
-                let mut err = self.span_fatal_err(self.span, Error::UselessDocComment);
+                let current_span = self.span;
+                let current_token = self.this_token_to_string();
+                let mut err = self.span_fatal_err(current_span, Error::UselessDocComment);
+
                 self.bump(); // consume the doc comment
                 if self.eat(&token::Comma) || self.token == token::CloseDelim(token::Brace) {
                     err.emit();
                 } else {
-                    return Err(err);
+                    let mut comma_err = self.span_fatal_help(current_span,
+                                    &format!("expected `,`, or `}}`, found `{}`", current_token),
```

Comment:
`current_token` could be a really long doc string, use instead the token description.